app/gui/src/controller/searcher/component/builder.rs

```diff
@@ -125,6 +138,8 @@ impl List {
     }
 
     /// Set the favorites in the list. Components are looked up by ID in the suggestion database.
+    /// When [`build`]ing a [`component::List`], only [`Component`]s with IDs passed to [`extend`]
+    /// will be retained (see the documentation of the [`build`] method).
```

Comment:
IMO this is kind of magic. I would never expect my function "extend" to influence what would be retained in a favourites group. The names of the functions need to be updated to reflect what they really do.